pyproject.toml

```diff
@@ -33,18 +33,18 @@ dependencies = [
     "numpy>=1.24",
     "pandas>=2.0",
     "xarray>=2023.04",
-    "netCDF4",
+    "netCDF4>=1.7",
     "packaging",
 ]
 dynamic = ["version"]
 
 [project.optional-dependencies]
 all = [
-    "contextily",
-    "geopandas",
-    "IPython",  # 'ipython' is not the correct module name.
-    "pyarrow",
-    "rioxarray",
+    "contextily>=1",
+    "geopandas>=0.14",
+    "IPython>=8",  # 'ipython' is not the correct module name.
+    "pyarrow>=14",
+    "rioxarray>=0.14",
```

Comment:
Setting lower bounds here, otherwise `uv` will try to install super old versions that may not work anymore (e.g. it tried to install rioxarray=0.0.1 at https://github.com/GenericMappingTools/pygmt/actions/runs/11945100244/job/33297179010#step:7:45), some which actually go back to Python 2! Happy to reconsider some of these version pins if needed.